scripts/package/README.md

```diff
@@ -34,44 +18,71 @@ Blockly.inject('blocklyDiv', {
 
 For samples on how to integrate Blockly into your project, view the list of samples at [blockly-samples](https://github.com/google/blockly-samples).
 
-### Importing Blockly
+## Installation
 
-When you import Blockly with `import * as Blockly from 'blockly';` you'll get the default modules:
-Blockly core, Blockly built-in blocks, the JavaScript generator and the English lang files.
+You can install this package either via `npm` or `unpkg`.
 
-If you need more flexibility, you'll want to define your imports more carefully:
+### unpkg
 
-#### Blockly Core
+```html
+<script src="https://unpkg.com/blockly/blockly.min.js"></script>
+```
+
+When importing from unpkg, you can access imports from the global namespace.
 
 ```js
-import * as Blockly from 'blockly/core';
+// Access Blockly.
+Blockly.thing;
+// Access the default blocks.
+Blockly.Blocks['block_type'];
+// Access the javascript generator.
+javascript.javascriptGenerator;
 ```
 
-#### Blockly built in blocks
+### npm
+
+```bash
+npm install blockly
+```
+
+## Imports
+
+Note: Using import of our package targets requires you to use a bundler (like webpack), since Blockly is packaged as a UMD, rather than an ESM.
 
 ```js
+// Import Blockly core.
+import * as Blockly from 'blockly/core';
+// Import the default blocks.
 import * as libraryBlocks from 'blockly/blocks';
+// Import a generator.
+import {javascriptGenerator} from 'blockly/javascript';
+// Import a message file.
+import * as En from 'blockly/msg/en';
 ```
 
-#### Blockly Generators
-
-If your application needs to generate code from the Blockly blocks, you'll want to include a generator.
+## Requires
 
 ```js
-import {pythonGenerator} from 'blockly/python';
+// Require Blockly core.
+const Blockly = require('blockly/core');
+// Require the default blocks.
+const libraryBlocks = require('blockly/blocks');
+// Require a generator.
+const {javascriptGenerator} = require('blockly/javascript');
+// Require a message file.
+const En = require('blockly/msg/en');
 ```
 
-to include the Python generator. You can also import `{javascriptGenerator} from 'blockly/javascript'`, `{phpGenerator} from 'blockly/php'`, `{dartGenerator} from 'blockly/dart'` and `{luaGenerator} from 'blockly/lua'`.
+## Applying blocks and messages
 
-#### Blockly Languages
+Once you have the library blocks and the message files, you also need to apply
+them.
 
 ```js
-import * as Fr from 'blockly/msg/fr';
-Blockly.setLocale(Fr);
+Blockly.Blocks = libraryBlocks;
+Blockly.setLocal(En);
```

Comment:
Still not sure whether this is necessary or not.

Reply:
```suggestion
Blockly.setLocale(En);
```
At the moment[*] mearly `import`ing (or `require`ing) `blockly/blocs` is sufficient to populate `Blockly.Blocks`, and trying to assign to that property does not change the dictionary object Blockly uses to look up block definitions, it merely causes it to no longer accessible to the developer (so they can't add custom block definitions or delete ones we've provided).

[*]: N.B. this is proposed to change—see #5875 and recent conversations re: best practices for publishing fields and blocks—but remains the case for the time being, and when it does change attempting to assign `Blockly.Blocks = /*...*/` will still be wrong.

As for whether the `setLocale` call (note final 'e') is required: for historical reasons this depends on whether a `Blockly` global variable is defined (and is pointing at the correct object).  If it is then the `setLocale` call is is not necessary; if it is not then it is.  To clarify further:

* When using a `<script>` tag to load `blockly_compressed.js` then `globalThis.Blockly` is defined and so merely loading `en.js` (as a script _or_ as a module) will set all the `Blockly.Msg[]`es.
  * A special case of this is loading `blockly.min.js` as a script, which includes both `blockly_compressed.js` and `en.js`
* When using `import` or `require`, `globalThis.Blockly` will not normally be defined by the UMD wrappers, but a developer could do this manually, e.g.:
  ```JS
  Blockly = require('blockly');  // N.B. no `const` so this sets a global.
  require('blockly/msg/en');  // Loads `Blockly.Msg` with no call to setLocale
  ```
  (This can also be done using `import` although it's harder to do by accident.)

Reply:
At the risk of getting lost in the weeds: it turns out the example I gave above is wrong:

Reply:
Ok so it sounds like:
  * For blocks, don't need to be assigned at the moment.
  * For languages, you do need to call `setLocale` unless you're loading from a script tag.

Reply:
I concur.